posixlib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala

```diff
@@ -47,25 +46,25 @@ object SocketHelpers {
 
         fcntl(sock, F_SETFL, O_NONBLOCK)
 
-        // Zone.alloc is documented as returning zeroed memory.
-        val fdsetPtr = alloc[fd_set] //  No need to FD_ZERO
+        // stackalloc is documented as returning zeroed memory
+        val fdsetPtr = stackalloc[fd_set] //  No need to FD_ZERO
```

Comment:
See comment below about drawbacks of largish allocations on the stack.
There the 1 KiB is evident and clearly (to me) largish.

Here the size of the fd_set allocation is most probably 128 bytes or more.
Whether that is 'largish' depends upon and may vary with the designer
and, ultimately, bug reports from active use in the field.

I myself would consider that largish and retain the heap `alloc`, along with a terse comment.   Other designers might have a different threshold for 
'largish'

The dominating cost of  this function are the trips across the wire in
`connect` and `select` (yuch!).   Running out of stack is a hard to discover
& fatal error.  Changing from heap to stack allocation here saves a few CPU
cycles, but the ratio of saved_Cycles/total_function_Cycles asymptotically approaches zero.